include/pmtiles.h

```diff
@@ -0,0 +1,57 @@
+/*! \file */ 
+#ifndef _PMTILES_H
+#define _PMTILES_H
+
+#include <fstream>
+#include "external/pmtiles.hpp"
+
+struct TileOffset {
+	uint64_t offset : 40;
+	size_t length : 24;
+
+	TileOffset(uint64_t offset, size_t length) : offset(offset),length(length) { }
+	TileOffset();
+};
+
+// Maximum number of tiles in a leaf directory
+#define LEAF_DIRECTORY_SIZE 10000000
+// Combined size of header and root directory (= start of tile data)
+#define HEADER_ROOT 16384
+// Tile ID at which to start using leaf directories (=z6/0/0)
```

Comment:
hardcoding this might not be useful in the case where you have a deep tileset that still happens to fit completely within the root directory - ideal for city-sized extracts to remove the overhead of the additional fetch